src/ucp/api/ucp_def.h

```diff
@@ -771,5 +771,103 @@ typedef struct ucp_ep_params {
  */
 #define UCP_ENTITY_NAME_MAX 32
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief The @ref ucp_transports_t and @ref ucp_transport_entry_t structures
+ *        are used when @ref ucp_ep_query is called to return an array of
+ *        transport name and device name pairs that are used by an active
+ *        endpoint.
+ *
+ *        The ucp_transport_t structure specifies the characteristics of the
+ *        ucp_transport_entry_t array.
+ *
+ *        The caller is responsible for the allocation and de-allocation
+ *        of the ucp_transport_entry_t array.
+ *
+ * Example: Implementation of a function to query the set of transport and
+ *          device name pairs used by the specified endpoint.
+ *
+ * @code{.c}
+ *   int query_transports(ucp_ep_h ep)
+ *   {
+ *     ucs_status_t status;
+ *     ucp_transport_entry_t *transport_entries;
+ *     ucp_ep_attr_t ep_attrs;
+ *
+ *     ep_attrs.field_mask = UCP_EP_ATTR_FIELD_TRANSPORTS;
+ *     ep_attrs.transports.entries = (ucp_transport_entry_t *)
+ *              malloc(10 * sizeof(ucp_transport_entry_t));
+ *     ep_attrs.transports.num_entries = 10;
+ *     ep_attrs.transports.entry_size = sizeof(ucp_transport_entry_t);
+ *     status = ucp_ep_query(ep, &ep_attrs);
+ *     if (status == UCS_OK) {
+ *         // ep_attrs.transports.num_entries = number of returned entries 
+ *         // ... process transport info ... 
+ *     }
+ *   }
+ * @endcode
+ */
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief A transport name and device name pair used by this endpoint. The
+ * caller is responsible for the allocation and deallocation of an array of
+ * these structures large enough to contain the desired number of transport
+ * and device name pairs.
+ *
+ * If new fields are added to this structure, they must be added at the end
```

Comment:
the "of this structure" at the end seems redundant, how about

Any new fields must be added to the end of this structure.

Reply:
@shamisp Since this pull request is merged and closed, how do I handle the change in comment wording? do I just make the change in #8116 which is still in process, create a new pull request, or do nothing?

Reply:
@drwootton I would create separate PR for documentation fixes.